src/extern/CProducer.cpp

```diff
@@ -127,6 +162,22 @@ int SendMessageSync(CProducer *producer, CMessage *msg, CSendResult *result) {
     return OK;
 }
 
+int SendMessageAsync(CProducer *producer, CMessage *msg, CSendSuccessCallback cSendSuccessCallback,CSendExceptionCallback cSendExceptionCallback){
+	if (producer == NULL || msg == NULL || cSendSuccessCallback == NULL || cSendExceptionCallback == NULL) {
+		return NULL_POINTER;
+	}
+	DefaultMQProducer *defaultMQProducer = (DefaultMQProducer *) producer;
+	MQMessage *message = (MQMessage *) msg;
+	CSendCallback* cSendCallback = new CSendCallback(cSendSuccessCallback , cSendExceptionCallback);
```

Comment:
`cSendCallback` should be deleted after `send`

Reply:
See MQClientAPIImpl::sendMessageAsync and AsyncCallbackWrap::operationComplete（90 line），and AsyncCallbackWrap::onException()[ 117 line]